In [17]:
import pandas as pd
import matplotlib.pyplot as plt   
%matplotlib inline
import statsmodels.api as sm
from sklearn.model_selection import train_test_split
import numpy as np
import statsmodels.formula.api as smf

In [18]:
df = pd.read_csv('salary.csv')

In [19]:
df = df.dropna(how='any',axis=0)

In [20]:
train, test = train_test_split(df, test_size=0.3)

In [21]:
X_train = train['yearsworked']
y_train = train['salary']

In [22]:
model = sm.OLS(y_train,X_train).fit()

In [23]:
print_model = model.summary()
print(print_model)

                            OLS Regression Results                            
Dep. Variable:                 salary   R-squared:                       0.756
Model:                            OLS   Adj. R-squared:                  0.756
Method:                 Least Squares   F-statistic:                     1111.
Date:                Thu, 28 Mar 2019   Prob (F-statistic):          8.90e-112
Time:                        06:26:18   Log-Likelihood:                -4157.4
No. Observations:                 359   AIC:                             8317.
Df Residuals:                     358   BIC:                             8321.
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                  coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------
yearsworked  2831.1430     84.950     33.327      

### Does the model significantly predicts the dependent variable? Report the amount of variance explained (R^2) and significance value (p) to support your answer.

The model significantly predicts the dependent variable. This is shown from variance (R^2) 0.766(higher value generally indicates a better fit) with a p value of 0.000... (less than 0.05 is considered to be statistically significant)


### What percentage of the variance in employees’ salaries is accounted for by the number of years they have worked?
76.5%



### What does the unstandardized coefficient (B or 'coef' in statsmodels) tell you about the relationship between Years Worked and Salary?
The coeffecient is 2840.5341 this means that when salary increases by 1, the number of salary increases by 2840.5341

### What do the 95% confidence intervals [0.025, 0.975] mean?

It shows the amount you would expect at a 95% confidence level. 

### Calculate the expected salary for someone with 12 years’ work experience.

In [27]:
model.predict(X_train[12])

446    19818.001214
86     11324.572122
87     19818.001214
214    42467.145459
242    62285.146674
302    53791.717582
327    65116.289704
22      2831.143031
331    82103.147888
110    25480.287276
469    36804.859398
373    84934.290919
35         0.000000
286    67947.432735
130    56622.860612
149    36804.859398
460    33973.716367
328    67947.432735
19      5662.286061
432    19818.001214
229    56622.860612
233    59454.003643
131    19818.001214
427        0.000000
122    33973.716367
75     14155.715153
174    16986.858184
198    56622.860612
179    25480.287276
187    31142.573337
           ...     
163    36804.859398
346    76440.861827
182     8493.429092
85     22649.144245
433    19818.001214
246    67947.432735
97     16986.858184
32         0.000000
513    70778.575766
375    62285.146674
4       5662.286061
275    65116.289704
269    62285.146674
442    22649.144245
119    16986.858184
88     31142.573337
41         0.000000
307    67947.432735
389     5662.286061


### Calculate the expected salary for someone with 80 years’ work experience. Are there any problems with this prediction? If so, what are they?

### We have only looked at the number of years an employee has worked. What other employee characteristics might influence their salary?

### How does your model compare when running it on the test set - what is the difference in the Root Mean Square Error (RMSE) between the training and test sets?

In [24]:
X_test = test['yearsworked']
y_test= test['salary']

In [25]:
test_model = sm.OLS(y_test,X_test).fit()

In [26]:
print_test_model = test_model.summary()
print(print_test_model)

                            OLS Regression Results                            
Dep. Variable:                 salary   R-squared:                       0.766
Model:                            OLS   Adj. R-squared:                  0.764
Method:                 Least Squares   F-statistic:                     499.7
Date:                Thu, 28 Mar 2019   Prob (F-statistic):           4.65e-50
Time:                        06:26:18   Log-Likelihood:                -1780.2
No. Observations:                 154   AIC:                             3562.
Df Residuals:                     153   BIC:                             3565.
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                  coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------
yearsworked  2953.0398    132.099     22.355      